### PR TITLE
Allow creating folders

### LINKED ISSUE #528: UI button to create folders
A more obvious way to create folders would be nice. The current way is not very intuitive.

Right now I create folders by naming a new password "folderName/passwordName".

### PR DESCRIPTION
**Pull request by msfjarvis**
## :loudspeaker: Type of change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description
Add a FAB transition with additional child FABs to provide new folder/new password actions. This is missing a LOT of things right now.
~Committing needs to be added~ (_no, you idiot_), and the theming needs some fixes.

## :bulb: Motivation and Context
Fixes #528

## :green_heart: How did you test it?
Manually

## :pencil: Checklist

- [x] I ran `./gradlew spotlessApply` before submitting the PR
- [x] I reviewed submitted code

### CODE REVIEW

**Comment by msfjarvis** [comment]
@Skrilltrax wanna take over this?

**Comment by Skrilltrax** [comment]
Sure

**Comment by msfjarvis** [comment]
> Sure

I'm doing some theming fixes right now, that's gonna leave committing and triggering a list refresh.

**Comment by Skrilltrax** [comment]
Oh okay, I'll just look at the current code until then

**Comment by msfjarvis** [comment]
I'm done now, you can take this up @Skrilltrax.

**Comment by Skrilltrax** [comment]
> I'm done now, you can take this up @Skrilltrax.

Working on it :+1:

**Comment by Skrilltrax** [comment]
@msfjarvis can you verify that the app is committing after creating folder ? Currently don't have any repo synced.

**Comment by msfjarvis** [comment]
It's not, that's the last blocker aside from any theming issues.

**Comment by Skrilltrax** [comment]
I think commit issue is because git won't add empty folder to unstaged changes automatically. Do we absolutely need to commit after creating a folder ?

**Comment by msfjarvis** [comment]
> I think commit issue is because git won't add empty folder to unstaged changes automatically. Do we absolutely need to commit after creating a folder ?

I don't think we even can. Massive brainfart there :man_shrugging:. This is good to go then?

File: app/src/main/java/com/zeapo/pwdstore/ui/dialogs/FolderCreationDialogFragment.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import com.google.android.material.textfield.TextInputEditText
+import com.zeapo.pwdstore.PasswordStore
+import com.zeapo.pwdstore.R
+import java.io.File
+
+class FolderCreationDialogFragment : DialogFragment() {
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val alertDialogBuilder = MaterialAlertDialogBuilder(requireContext())
+        alertDialogBuilder.setView(R.layout.folder_creation_dialog_fragment)
+        alertDialogBuilder.setPositiveButton("Create") { _: DialogInterface, _: Int ->
```
Comment (msfjarvis) [inline app/src/main/java/com/zeapo/pwdstore/ui/dialogs/FolderCreationDialogFragment.kt]: This needs to go in strings.xml

File: app/src/main/java/com/zeapo/pwdstore/ui/dialogs/FolderCreationDialogFragment.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    private fun createDirectory(currentDir: String) {
+        val dialog = requireDialog()
+        val materialTextView = dialog.findViewById<TextInputEditText>(R.id.folder_name_text)
+        val folderName = materialTextView.text.toString()
+        File("$currentDir/$folderName").mkdir()
+        (requireActivity() as PasswordStore).updateListAdapter()
+    }
+
+    companion object {
+        private const val TAG = "FolderCreationDialogFragment"
```
Comment (msfjarvis) [inline app/src/main/java/com/zeapo/pwdstore/ui/dialogs/FolderCreationDialogFragment.kt]: Don't think we ended up using this, fine with deleting

**Comment by Skrilltrax** [comment]
> Code's fine, just need to test at runtime before merging.

Works fine for me

**Pull request merged**